src/selectLap/laps.ts

```diff
@@ -31,19 +31,24 @@ function initializeAPI() {
 }
 
 initializeMap();
-initializeAPI();
 
-allFilesSubject$.subscribe((filenames) => {
-  if (!filenames) {
-    alert("Could not find filenames for races.");
-    hideSpinner();
-  } else {
-    for (const filename of filenames) {
-      allFilenames.push(filename);
-      signalNewLapsPerRunRequest$.next(filename);
+if (!passedFilename) {
+  initializeAPI();
+
+  allFilesSubject$.subscribe((filenames) => {
+    if (!filenames) {
+      alert("Could not find filenames for races.");
+      hideSpinner();
+    } else {
+      for (const filename of filenames) {
+        signalNewLapsPerRunRequest$.next(filename);
+        localStorage.setItem("Passed Filename", `${filename}`);
```

Comment:
I'm noticing that your key for all the local storage is "Passed Filename", so how will you know which filename to retrieve? I would advise using template literals for your key, so that its unique. Because I think currently you are replacing the filename... unless maybe i'm not understanding the logic of the code

Reply:
SO for Passed Filenames, when a certain race is selected that race's filename if passed to the next page where the laps for that filename can be fetched. And "Filenames" are supposed to be an array of strings containing all the filenames.

Reply:
Understand now what you were saying and have implemented the fix.

Reply:
Oh okay cool, I saw 👏🏾✨. Sorry I was out earlier I am only seeing your responses now